core/src/main/java/tc/oc/pgm/spawner/Spawner.java

```diff
@@ -102,27 +101,27 @@ public void onItemMerge(ItemMergeEvent event) {
       return;
     }
 
-    int entitySpawnerID = -1;
-    int targetSpawnerID = -1;
+    String entitySpawnerId = "";
+    String targetSpawnerId = "";
     if (event.getEntity().hasMetadata(METADATA_KEY)) {
-      entitySpawnerID =
-          MetadataUtils.getMetadata(event.getEntity(), METADATA_KEY, PGM.get()).asInt();
+      entitySpawnerId =
+          MetadataUtils.getMetadata(event.getEntity(), METADATA_KEY, PGM.get()).toString();
     }
     if (event.getTarget().hasMetadata(METADATA_KEY)) {
-      targetSpawnerID =
-          MetadataUtils.getMetadata(event.getTarget(), METADATA_KEY, PGM.get()).asInt();
+      targetSpawnerId =
+          MetadataUtils.getMetadata(event.getTarget(), METADATA_KEY, PGM.get()).toString();
     }
     // Cancel the merge if the items are from different PGM spawners
-    if (entitySpawnerID != targetSpawnerID) {
+    if (!entitySpawnerId.equals(targetSpawnerId)) {
       event.setCancelled(true);
       return;
     }
```

Comment:
Whole logic of this event handler seems rather superfluous, including the code above. Can be simplified:
```
    boolean entityTracked = event.getEntity().hasMetadata(METADATA_KEY);
    boolean targetTracked = event.getTarget().hasMetadata(METADATA_KEY);
    if (!entityTracked && !targetTracked) return; // None affected
    if (entityTracked && targetTracked) {
      String entitySpawnerId =
          MetadataUtils.getMetadata(event.getEntity(), METADATA_KEY, PGM.get()).toString();
      String targetSpawnerId =
          MetadataUtils.getMetadata(event.getTarget(), METADATA_KEY, PGM.get()).toString();
      if (entitySpawnerId.equals(targetSpawnerId)) return; // Same spawner, allow merge
    }
    event.setCancelled(true);
```